dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -610,13 +614,40 @@ private void writeRelationAnnotation(BratAnnotationDocument aDoc, FeatureStructu
         }
     }
 
+
+    
+    private BratTextAnnotation splitNewline(AnnotationFS aFS)
+    {
+
+        // extract all but newlines as groups
+        Matcher m = newlineExtractPattern.matcher(aFS.getCoveredText());
+        // counts the number of matches to initialize arrays sized
```

Comment:
Instead of maintaining separate arrays for begin and end offsets, I would recommend to introduce an `Offsets` class which contains begin and end offsets and then to use a `List<Offsets>`. That would also remove the need for passing twice through the pattern here as you can construct the list in one go.